tpot/operators/classifiers/base.py

```diff
@@ -57,8 +56,22 @@ def _train_and_predict(self, input_matrix, operator_args):
         clf.fit(self.training_features, self.training_classes)
 
         all_features = np.delete(input_matrix, non_feature_columns, axis=1)
-        input_matrix[:, GUESS_COL] = clf.predict(all_features)
+        if self.clf_eval_func == 'decision_function':
+            if 'decision_function' in dir(clf):
+                input_matrix = np.insert(input_matrix, GUESS_COL, clf.decision_function(all_features).T, axis=1)
```

Comment:
If you need to store more data you'll probably be best off adding another "non_feature_column" entirely - not just keeping a reference to a column after `GUESS_COL`. I never really considered that contingency, though, so that might break all other code dealing with column indices.